test/e2e/storage/volume_provisioning.go

```diff
@@ -461,10 +460,9 @@ var _ = SIGDescribe("Dynamic Provisioning", func() {
 				framework.ExpectNoError(framework.DeletePersistentVolumeClaim(c, pvc.Name, ns), "Failed to delete PVC ", pvc.Name)
 			}()
 
-			// The claim should timeout phase:Pending
-			err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, c, ns, pvc.Name, 2*time.Second, framework.ClaimProvisionTimeout)
-			Expect(err).To(HaveOccurred())
-			framework.Logf(err.Error())
+			// The claim should receive a ProvisioningFailed event with message mentioning the unmanaged zone
+			err = framework.WaitForParticularEvent(c, ns, pvc, events.ProvisioningFailed, "kubernetes does not manage zone")
```

Comment:
Are there windows where the even occurs but `WaitForParticularEvent` misses it? Or the even does not occur because the the controller reconciler bypasses this particular even to get to the desired state?